Lesson_10/wineshop_dyn_amounts.html

```diff
@@ -0,0 +1,189 @@
+            }
+
+            let ratingHtml = '<span class="fa fa-star checked"></span>'.repeat(wines[i].rating)
+                + '<span class="fa fa-star"></span>'.repeat(5 - wines[i].rating);
+
+            let amountWithComment = getAmountWithComment(wines[i].amount);
+
+            wineListEl.innerHTML = wineListEl.innerHTML + `
+            
+            <div class="wineCard">
+                <div class="wineImage">
+                    <img class="imageOfWine" src="${wines[i].image}" alt="Image of ${wines[i].title}">
+                </div>
+                <div class="wineProperties">
+                    <span class="wineTitle">${wines[i].title}</span> <br>
+                    <b>Type</b>: ${wines[i].type}<br>
+                    <b>Origin</b>: ${wines[i].origin.country}/${wines[i].origin.region}/${wines[i].origin.winery} <br>
+                    <b>Year</b>: ${wines[i].year} <br>
+                    <b>Grapes</b>: ${wines[i].grapes} <br>
+                    <b>Price</b>: ${wines[i].price} ${wines[i].currency} <br>   
+                    <b>Amount</b>:
+                    <span id="stockAmount${i}">${amountWithComment}</span>
+                    <p class="starRating">${ratingHtml} </p>
+                    <p><button class="wineBtn" id="addWineBtn${i}">Add to cart</button></p>
+                </div>
+            </div> 
+            `;
+        }
+
+        for (let i = 0; i < wines.length; i++) {
+            const addWineBtn = document.getElementById('addWineBtn' + i);
+            addWineBtn.addEventListener("click", function () { addWineToCart(i); });
+            if (wines[i].amount < 1) {
+                addWineBtn.disabled = true;
+            }
+        }
+
+        function getAmountWithComment(amount) {
+            let amountComment = '';
+            if (amount < 1) {
+                amountComment = ' - out of stock';
+            } else if (amount === 1) {
+                amountComment = ' - last chance';
+            } else if (amount <= 3) {
+                amountComment = ' - almost out of stock';
+            }
+            return amount + ' ' + amountComment;
+        }
+
+        function addWineToCart(wId) {
+            shoppingCart.push(
+                {
+                    title: wines[wId].title,
+                    price: wines[wId].price
+                });
+            wines[wId].amount = wines[wId].amount - 1;
+            const stockAmountEl = document.getElementById('stockAmount' + wId);
+            stockAmountEl.textContent = getAmountWithComment(wines[wId].amount);
+            if (wines[wId].amount < 1) {
+                const addWineBtn = document.getElementById('addWineBtn' + wId);
+                addWineBtn.disabled = true;
+            }
+
+
+        };
+
+        function showShoppingCartProducts() {
+            let totalPrice = 0;
+            const shoppingCartEl = document.getElementById('shoppingCart');
+            for (let i = 0; i < shoppingCart.length; i++) {
```

Comment:
before for-loop you need to clear innerHTML of your shoppint cart, because if user clicks more and more on Show shopping cart button the content would be duplicated.

Reply:
Thanks! Done 
shoppingCartEl.innerHTML = "";